transcom/mymove
# created checkValidSitAddlDates and added to payment request creation

## [B-18522](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A895040)

## Summary

Introduces new payment request creator rule to validate the dates on Sit Additional Days payment request creation.

When the rule detects an issue, the handler will return a 422 (Unprocessable Entity) response

## Verification Steps for Reviewers

These are to be checked by a reviewer.


### How to test

1. Login as the prime simulator
2. Select any move
3. Create new HHG shipment, filling fields as necessary
4. Scroll down to your new HHG shipment
5. Select “Add Service Item”
6. Create a new “Destination SIT” service item
7. Select “Create Payment Request”
8. Scroll down to your new HHG shipment - Make sure it’s the correct one!
9. Once you have found your HHG shipment, scroll down its options of shipment service items until you see “Domestic destination add’l SIT”
10. Select “add to payment request” and provide a payment start date that comes AFTER the payment end date
11. Submit payment request
12. You should receive a detailed error

## Screenshots

![image](https://github.com/transcom/mymove/assets/136814362/42e54e77-b941-4c56-9495-c1d4cd80c53c)
![image](https://github.com/transcom/mymove/assets/136814362/d187b802-50b4-4ab9-bcc9-4d8df4d2c35d)
